### PR TITLE
chore(otel): collect telemetry for otel configurations

### PR DESCRIPTION
**Pull request by mabdinur**
## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by avara1986 — APPROVED** [review]

**Comment by erikayasuda** [comment]
> `CODEOWNERS` have been resolved as:
> 
> ```
> ddtrace/settings/_otel_remapper.py                                      @DataDog/apm-core-python
> tests/.suitespec.json                                                   @DataDog/python-guild @DataDog/apm-core-python
> tests/opentelemetry/test_config.py                                      @DataDog/apm-sdk-api-python
> tests/telemetry/test_writer.py                                          @DataDog/apm-core-python
> ```

@mabdinur Should `ddtrace/settings/_otel_remapper.py` be owned by `@DataDog/apm-sdk-api-python`?

**Review by emmettbutler — APPROVED** [review]

**Comment by mabdinur** [comment]
> > `CODEOWNERS` have been resolved as:
> > ```
> > ddtrace/settings/_otel_remapper.py                                      @DataDog/apm-core-python
> > tests/.suitespec.json                                                   @DataDog/python-guild @DataDog/apm-core-python
> > tests/opentelemetry/test_config.py                                      @DataDog/apm-sdk-api-python
> > tests/telemetry/test_writer.py                                          @DataDog/apm-core-python
> > ```
> 
> @mabdinur Should `ddtrace/settings/_otel_remapper.py` be owned by `@DataDog/apm-sdk-api-python`?

Yup I already update suitespec: https://github.com/DataDog/dd-trace-py/pull/11351/files#diff-fc9f8994ea260e13b138fb9cd173abaa897c4a2d099e87ee6c0118f0be3be822

**Comment by mabdinur** [comment]
I hate molton 😞

File: tests/telemetry/test_writer.py [hunk 1]
```diff
@@ -813,3 +813,64 @@ def test_telemetry_writer_is_using_agent_by_default_if_api_key_is_not_available(
...
+    tags = [m["tags"] for m in env_hiding_metrics]
+    assert tags == [["config_opentelemetry:otel_service_name", "config_datadog:dd_service"]]
+
+    env_unsupported_metrics = test_agent_session.get_metrics("otel.env.unsupported")
+    tags = [m["tags"] for m in env_unsupported_metrics]
+    assert tags == [["config_opentelemetry:otel_unsupported_config"]]
+
+    env_invalid_metrics = test_agent_session.get_metrics("otel.env.invalid")
+    tags = [m["tags"] for m in env_invalid_metrics]
+    assert tags == [["config_opentelemetry:otel_logs_exporter", "config_datadog:"]]
```
Comment (zacharycmontoya) [inline tests/telemetry/test_writer.py:876]: I don't think the `config_datadog` tag is supposed to be present, can you verify this?

**Review by zacharycmontoya — APPROVED** [review]
LGTM from the API side

**Pull request merged**